mood-analysis.rb

```diff
@@ -19,6 +20,55 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def get_sanitized_input(words)
+  # input now matches feelings hash 
+  words.downcase!
+  # did not write a strip_punctuation method but this line strips it
+  words.gsub!(/[^a-z0-9\s]/i, '')
+  return words
+end 
+
+def get_happy_days(emoji_array)
+  happy = 0 
+  smile = ":-)"
+
+  number_of_days = 1 
+  emoji_array.each do |emoji|
+    if emoji == smile
+      happy += 1
+      if happy >= 3 && happy < 4
+        puts "It has taken #{number_of_days} days for #{happy} happy days to occur"
+      end
+    end
+    number_of_days += 1 
+  end
+  if happy > 3
+    puts "There have only been #{happy} happy days out of a total of #{number_of_days-1} days"
+  end 
+end
+
+def get_overall_mood(emoji_array)
+  overall_mood = 0 
+  smile = ":-)"
+  frown = ":-("
+  meh = ":-|"
+
+  emoji_array.each do |emoji|
+    if emoji == smile
+      overall_mood += 1
+    else emoji == frown
+      overall_mood -= 1
+    end 
+  end 
+  if overall_mood > 0
+    puts "overall mood is #{smile}"
+  elsif overall_mood < 0 
+    puts "overall mood is #{frown}"
+  else 
+    puts "overall mood is #{meh}"
+  end 
+end 
```

Comment:
Hrm. This logic is flawed. I believe this doesn't work if the :-| is the highest mood. It actually only gives you meh if smile and frown are the same number, regardless of how many meh's there are.

This method needs complete overhaul.